artemis-server/src/test/java/org/apache/activemq/artemis/core/server/impl/ActiveMQServerImplTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.core.server.impl;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.activemq.artemis.core.server.ActiveMQServer;
+import org.apache.activemq.artemis.utils.ThreadLeakCheckRule;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ActiveMQServerImplTest {
+
+   @Test
+   public void testScheduledPoolGC() throws Exception {
+      ActiveMQServer server = new ActiveMQServerImpl();
+
+      server.start();
+
+      WeakReference<Runnable> scheduledRunnableRef;
+      {
+         Runnable scheduledRunnable = new Runnable() { @Override public void run() { } };
+         scheduledRunnableRef = new WeakReference<>(scheduledRunnable);
+
+         ScheduledFuture scheduledFuture = server.getScheduledPool().schedule(scheduledRunnable, 5000, TimeUnit.MILLISECONDS);
+
+         scheduledFuture.cancel(true);
+      }
+
+      Assert.assertNotEquals(null, scheduledRunnableRef.get());
```

Comment:
GC is not precise (when things got out of scope/not referenced they got collected immediately) and depend on the type of GC chosen: probably you need to issue (full) GC and right after check the weak reference, but it is not reliable enough. We have other tests that do something similar using an ad-hoc call to GC, will point you them on Monday